### PR TITLE
add support for "clips" / trimming videos

### PR DESCRIPTION
**Pull request by dumbmoron**

### CODE REVIEW

File: src/modules/stream/cut.js [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+    if (format?.includes('mp4'))
+        return 'mp4';
+
+    throw new Error(`unknown format: ${format}`);
+}
+
+const close_enough = (a, b) => Math.abs(a - b) < 0.02;
+const last = arr => arr[arr.length - 1];
+const output = process => last(process.stdio);
+const lerp = (a, b, α) => Number(a) + α * (b - a);
```
Comment (wukko) [inline src/modules/stream/cut.js:20]: remove greek letters please

File: src/modules/stream/cut.js [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+
+function getKeyframesAround(url, { start, end }) {
+    start = Math.floor(start), end = Math.ceil(end);
+    assert(end > start);
+
+    const ε = 30;
```
Comment (wukko) [inline src/modules/stream/cut.js:26]: here too

File: src/modules/stream/cut.js (hunk 1 above)
Comment (princessmortix) [inline src/modules/stream/cut.js:20]: I believe you should use egypt letters too

File: src/modules/stream/cut.js [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+    if (format?.includes('webm'))
+        return 'webm';
+
+    if (format?.includes('mp4'))
+        return 'mp4';
+
+    throw new Error(`unknown format: ${format}`);
+}
+
+const close_enough = (a, b) => Math.abs(a - b) < 0.02;
```
Comment (synzr) [inline src/modules/stream/cut.js:17]: eww snake case in javascript

File: src/modules/stream/cut.js (hunk 1 above)
Comment (synzr) [inline src/modules/stream/cut.js:20]: eww snake case in javascript

**Comment by maherbel** [comment]
Is this feature canceled ?

**Comment by dumbmoron** [comment]
> Is this feature canceled ?

no but it needs to get rewritten because this implementation was kinda half-working

**Comment by dhawansolanki** [comment]
How to actually run this feature as implemented in the above commits?

**Pull request closed without merging**